Sources/OpenAPIRuntime/Base/Acceptable.swift

```diff
@@ -0,0 +1,167 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// The protocol that all generated `AcceptableContentType` enums conform to.
+public protocol AcceptableProtocol: RawRepresentable, Sendable, Hashable, CaseIterable where RawValue == String {}
+
+/// A quality value used to describe the order of priority in a comma-separated
+/// list of values, such as in the Accept header.
+public struct QualityValue: Sendable, Hashable {
+
+    /// As the quality value only retains up to and including 3 decimal digits,
+    /// we store it in terms of the thousands.
+    ///
+    /// This allows predictable equality comparisons and sorting.
+    ///
+    /// For example, 1000 thousands is the quality value of 1.0.
+    private let thousands: UInt16
+
+    /// Creates a new quality value of the default value 1.0.
+    public init() {
```

Comment:
Without reading the docs it's not obvious that `let q = QualityValue()` would default to 1.0. From a readability perspective it might make more sense to not have a default init.

Reply:
It is the default according to the RFC. Are you saying the user should be forced to always specify quality value, even if they don't want to customize it? That might be strange, as `application/json, text/plain` and `application/json; q=1.0, text/plain; q=1.0` are completely equivalent, and unless people customize the quality value to be anything other than 1.0, they usually don't explicitly specify it.

Reply:
No, I would just expect it to be defaulted at a higher level like how it is for `AcceptHeaderContentType`.

Reply:
Oh sure, turns out we're only using it in tests. I'll remove it.

Reply:
Removed in https://github.com/apple/swift-openapi-runtime/pull/37/commits/068b34e4e9da4faa920e692f46ab1b77fcc69cef